### PR TITLE
tools/zep_dispatch: enhancements and fixes

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This is a collection of improvements I added while working on the CoAP multicast implementation.

#### topogen

 - binary mode: links will be dropped below a cutoff-point, above it they are always successful
 - numeric node names: allows for easier scripting
 - make link quality drop off quadratically with distance
 - always place the first node at a known location

#### dispatcher

 - fix bug where packet won't get delivered to any node after it was dropped on a single link
 - count rx/tx frames per node
 - add possibility to store pidfile

### CODE REVIEW

**Comment by kfessel** [comment]
could you sketch a Testing procedure

**Review by kfessel — COMMENTED** [review]
just some comments

File: dist/tools/zep_dispatch/topogen.c [hunk 1]
```diff
@@ -51,7 +72,7 @@ static double node_distance(const struct node *a, const struct node *b)
 
 static double node_distance_weight(const struct node *a, const struct node *b)
 {
-    double w = 1 - node_distance(a, b) / a->r;
+    double w = 1 - pow(node_distance(a, b), 2) / pow(a->r, 2);
```
[the remark this answers is not in the document]
Comment (benpicco) [reply dist/tools/zep_dispatch/topogen.c:77]: tbh I don't think there is currently anyone besides me using this

File: dist/tools/zep_dispatch/topogen.c [hunk 2]
```diff
@@ -13,6 +13,10 @@
 #include <math.h>
 #include <unistd.h>
 
+#ifndef CONFIG_USE_NUMERIC_NAMES
+#define CONFIG_USE_NUMERIC_NAMES 1
```
[the remark this answers is not in the document]
Comment (benpicco) [reply dist/tools/zep_dispatch/topogen.c:17]: Hm, I wasn't sure if I shouldn't remove it altogether.
The alphabetic names look nice for node numbers < 26, but once we use two letters, it's much more confusing than just numbering them.

**Review by kfessel — APPROVED** [review]
I think benpicco tested this well.
Since benpicco  is the main author of this application and I can't see special/strange/surprising changes just enhancements and a slight change in distance calculation.

please squash and merge these.

File: dist/tools/zep_dispatch/topogen.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (kfessel) [reply dist/tools/zep_dispatch/topogen.c:17]: Lets keep them in hidden

**Comment by OlegHahm** [comment]
I would be interested at some point to discuss whether this tooling and des-virt can be merged somehow. It's a bit weird that we have (at least) two different network emulators in the repository - and apparently both them are used by only very few people.

**Pull request merged**